tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -256,36 +256,101 @@ class UnsortedSegmentSumOp : public OpKernel {
     Tensor* output = nullptr;
     OP_REQUIRES_OK(context, context->allocate_output(0, output_shape, &output));
     auto output_flat = output->flat_outer_dims<T>();
-    output_flat.setZero();
-
+    if(is_max_){
```

Comment:
Couple of nits regarding style, to be consistent with the rest of the file:

Add space after "if" and ")"
if (is_max_) {